packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js

```diff
@@ -1058,4 +1060,25 @@ module.exports.external = [
       return PERSONAL_RELATIONSHIP_TYPE_CODE;
     },
   },
+  {
+    name: "decodeHtmlEntities",
+    description: "Decodes html strings",
+    func: function (str) {
+      if (str === undefined || str === null) {
+        return "";
+      }
+      const result = he.decode(str);
+      return result;
+    },
+  },
+  {
+    name: "removeNonNumeric",
+    description: "Removes all non-numeric characters from a string, keeping decimal points",
+    func: function (str) {
+      if (str === undefined || str === null) {
+        return "";
+      }
+      return str.replace(/[^0-9.]/g, '');
+    },
```

Comment:
Maybe make this `toNumber`, which would remove non-numeric chars as above and try to convert it to `Number()`, and if it results in `NaN` then we return empty?

Reply:
There was a bit of a refactor here. I think that for HAPI this should stay as a string though

Reply:
We can convert back to string, obviously, but IMO we shouldn't try to implement number conversion if `Number()` already does it for us - or at least try to use as much of it as possible.

For example, if we process `0.255e3` through this code we'll result in a totally different number than if we treat it as an exponential.

Reply:
Right so the new extractDecimal function runs the string through the exact fhir definition of decimal. I am not going to use Number because the definition of decimal are slightly different. All decimals are numbers, but not all numbers are decimals:

`const decimalRegex = /^-?(?:0|[1-9][0-9]*)(?:\.[0-9]+)?(?:[eE][+-]?[0-9]+)?$/;

const testCases = [
    "   42 ",  // Leading and trailing whitespaces
    "",        // Empty string
    "NaN",     // Special numeric value NaN
    "Infinity",// Special numeric value Infinity
    "0xFF"     // Hexadecimal number
];

testCases.forEach(value => {
    const regexMatch = decimalRegex.test(value.trim());
    const numberResult = Number(value);
    console.log(`Value: ${value}, Regex Match: ${regexMatch}, Number Result: ${numberResult}`);
});
`
```
Value:    42 , Regex Match: true, Number Result: 42
Value: , Regex Match: false, Number Result: 0
Value: NaN, Regex Match: false, Number Result: NaN
Value: Infinity, Regex Match: false, Number Result: Infinity
Value: 0xFF, Regex Match: false, Number Result: 255
```